### ISSUE TITLE
"Live Chat" tabs are no longer specific to an endpoint

### ISSUE THREAD

**Issue by v-kydela** [microsoft/BotFramework-Emulator#2121]
## Version
Latest: 4.8.1

## Describe the bug
When I open a bot file with multiple endpoints, the bot conversation tab just says "Live Chat" and gives no indication of what endpoint the conversation is associated with. Also, clicking an endpoint will now always open a new tab instead of focusing on an existing tab if a tab for that endpoint is already open.

## To Reproduce
1. Open a bot configuration
2. Click on an endpoint in the endpoints list multiple times
3. See multiple tabs open that don't say what their endpoint is

## Expected behavior
Clicking on an endpoint should focus on an existing tab if a tab for that endpoint is already open, and the tab should say what its endpoint is.

## Screenshots
![](https://i.imgur.com/7MRNKZ1.gif)

## Additional context
This bug may have come about as an attempt to fix this problem: https://github.com/microsoft/BotFramework-Emulator/issues/1611

[bug]

**Comment by tonyanziano** [comment]
This is something on our radar and we are tracking it at #2084 .

Closing this issue in favor of the aforementioned one.

**Comment by v-kydela** [comment]
@tonyanziano - That issue looks like it's just about displaying the endpoint in the UI of the tab. That's good, but are you also tracking the problem of the endpoint opening duplicate tabs?

**Comment by tonyanziano** [comment]
@v-kydela That issue is not tracking the behavior you are describing.

However, I'm not sure that's exactly a bug or incorrect behavior.

Most of the bots and samples I use are configured to run on the same url / port, `localhost:3978`. Let's say I was talking to an echo bot on  `localhost:3978`, and then stopped that bot and started running my To-Do bot at the same endpoint and connected to that. I wouldn't find it very useful for the Emulator to re-open the tab used for my echo bot, wiping the conversation history, and using it for my completely unrelated bot.

I think this would make sense in the context of having a .bot file open and clicking the same endpoint in the endpoint explorer, but less for connecting to a bot via URL.

![image](https://user-images.githubusercontent.com/3452012/77808175-9996b080-7047-11ea-93e2-d45b53a8820b.png)

**Comment by v-kydela** [comment]
@tonyanziano - I was under the impression that there is no endpoints list when you connect to a bot via URL.

Anyway, the reason I think this is a bug is because it's new behavior. The discussion in this issue made it seem like the old behavior of having one tab for one endpoint was desirable: https://github.com/microsoft/BotFramework-Emulator/issues/1611

**Comment by tonyanziano** [comment]
That is correct. There is no endpoints list when you connect to a bot via URL. The list only shows when you have a .bot file open in the Emulator.

This however, is not new behavior in the scenario of opening a livechat via URL. The old code that used the `focusExistingChatIfAvailable` flag -- which is referred to in the discussion you linked -- was only ever called in the code path for opening endpoints via .bot file.

So we can reopen this and use it to track restoring the behavior to the .bot file scenario, but I don't think this should be the behavior in the standalone URL scenario.

**Comment by v-kydela** [comment]
@tonyanziano - I'm sorry, I'm still not understanding what you mean. How could this possibly be the behavior for the standalone URL scenario if the standalone URL scenario doesn't have an endpoints list? The behavior I'm describing is the behavior of the endpoints list.

Anyway I'll reopen this since I have your permission, thank you

**Comment by tonyanziano** [comment]
Ah, that's my bad. I completely misread your issue.

Then ignoring what I said about the open via URL scenario, we are in agreement.

**Comment by cwhitten** [comment]
Note to implementer, reference https://github.com/microsoft/BotFramework-Emulator/issues/1611 and test scenario for regressions
